terminal-config/altapay_config.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+require_once(__DIR__ . '/altapay-libs/autoload.php');
+
+use Altapay\Api\Others\Terminals;
+use Altapay\Api\Test\TestAuthentication;
+use GuzzleHttp\Exception\ClientException;
+use Altapay\Authentication;
+
+// Settings
+$apiUser = '~gatewayusername~';
+$apiPass = '~gatewaypass~';
+$url = '~gatewayurl~';
+
+try {
+    $api = new TestAuthentication(new Authentication($apiUser, $apiPass, $url));
+    $response = $api->call();
+    if (!$response) {
+        echo "API credentials are incorrect";
+        exit();
+    }
+} catch (ClientException $e) {
+    echo "Error:" . $e->getMessage();
+    exit();
+} catch (Exception $e) {
+    echo "Error:" . $e->getMessage();
+    exit();
+}
+
+require_once(__DIR__ . '/admin/config.php');
+require_once(__DIR__ . '/system/startup.php');
+
+// Registry
+$registry = new Registry();
+// Loader
+$loader = new Loader($registry);
+$registry->set('load', $loader);
+// Config
+$config = new Config();
+$registry->set('config', $config);
+// Database
+$db = new DB(DB_DRIVER, DB_HOSTNAME, DB_USERNAME, DB_PASSWORD, DB_DATABASE, DB_PORT);
+$registry->set('db', $db);
+
+$loader->model('setting/setting');
+$loader->model('design/layout');
+$loader->model('extension/module/altapay');
+$loader->model('user/user_group');
+$modelUserGroup = new ModelUserUserGroup($registry);
+
+// API settings
+addSettingField($db, 'module_altapay', 'module_altapay_gateway_url', $url);
+addSettingField($db, 'module_altapay', 'module_altapay_gateway_username', $apiUser);
+addSettingField($db, 'module_altapay', 'module_altapay_gateway_password', $apiPass);
+addSettingField($db, 'module_altapay', 'module_altapay_status', 1);
+// Add currency field
+$currencyId = addCurrencyField($db, "Danish Krone", "DKK", "DKK");
+// Set currency
+addSettingField($db, 'config', 'config_currency', 'DKK');
+// Register module
+addExtensionField($db, 'module', 'altapay');
+
+// Create orders table
+$db->query("CREATE TABLE IF NOT EXISTS `" . DB_PREFIX . "altapay_orders` ( `altapay_order_id` int(11) NOT NULL AUTO_INCREMENT, `order_id` int(11) NOT NULL, `created` DATETIME NOT NULL, `modified` DATETIME NOT NULL, `amount` DECIMAL( 10, 2 ) NOT NULL, `currency_code` CHAR(3) NOT NULL, `transaction_id` VARCHAR(24) NOT NULL, `capture_status` INT(1) DEFAULT NULL, `void_status` INT(1) DEFAULT NULL, `refund_status` INT(1) DEFAULT NULL, PRIMARY KEY (`altapay_order_id`) ) ENGINE=MyISAM DEFAULT COLLATE=utf8_general_ci;");
+
+try {
+    $api = new Terminals(new Authentication($apiUser, $apiPass, $url));
+    $response = $api->call();
+    $i = 1;
+
+    foreach ($response->Terminals as $key => $terminal) {
+        if ($terminal->Country == 'DK') {
```

Comment:
What happens if the terminal country is not DK?

Reply:
@emicha For now, we are handling only a single store and defined the store currency to 'DKK'. If we are looking for a multi-store then we need to update the check otherwise for this scenario this will skip the terminal that has a country other than DK.